File: golang/q02/part1/part1.go
package part1

import (
	c "aoc2023/q02/common"
	"fmt"
)

var maxCubes c.Round = c.Round{Red: 12, Green: 13, Blue: 14}

func Part1(games []c.Game) {
	sum := 0

	for i, game := range games {
		gameMax := game.MaxColors()

		if gameMax.Red > maxCubes.Red || gameMax.Green > maxCubes.Green || gameMax.Blue > maxCubes.Blue {
			continue
		}

		sum += i + 1
	}

	fmt.Println("Part 1:", sum)
}

File: js/q02/part1/part1.js
import { parseFile, getGameMaxes } from "../common/index.js";

const maxCubes = {
  red: 12,
  green: 13,
  blue: 14,
};

function isValid(gameMaxes) {
  return gameMaxes.red <= maxCubes.red
    && gameMaxes.green <= maxCubes.green
    && gameMaxes.blue <= maxCubes.blue;
}

function part1(path) {
  const parseFileResult = parseFile(path);
  if (parseFileResult.isErr()) {
    console.error(`Parse file error: ${parseFileResult.unwrap()}`);
    return;
  }

  const validGames = parseFileResult.unwrap().filter(game => {
    const maxes = getGameMaxes(game);
    return isValid(maxes);
  });

  const idSum = validGames.reduce((acc, curr) => acc + curr.number, 0);
  console.log(`Part 1: ${idSum}`);
}

part1('./data/input.txt');
